integration_tests/src/main/python/join_test.py

```diff
@@ -33,12 +33,30 @@
         pytest.param(FloatGen(nullable=False), marks=[incompat]),
         pytest.param(DoubleGen(nullable=False), marks=[incompat])]
 
+basic_struct_gen = StructGen([
+    ['child' + str(ind), sub_gen]
+    for ind, sub_gen in enumerate([StringGen(), ByteGen(), ShortGen(), IntegerGen(), LongGen(),
+                                   BooleanGen(), DateGen(), TimestampGen(), null_gen, decimal_gen_default])],
+    nullable=False)
+
+basic_struct_gen_with_no_null_child = StructGen([
+    ['child' + str(ind), sub_gen]
+    for ind, sub_gen in enumerate([StringGen(nullable=False), ByteGen(nullable=False),
+                                   ShortGen(nullable=False), IntegerGen(nullable=False), LongGen(nullable=False),
+                                   BooleanGen(nullable=False), DateGen(nullable=False), TimestampGen(nullable=False)])],
+    nullable=False)
+
+basic_struct_gen_with_floats = StructGen([['child0', FloatGen()], ['child1', DoubleGen()]], nullable=False)
+
+struct_gens = [basic_struct_gen, basic_struct_gen_with_no_null_child]
+
 double_gen = [pytest.param(DoubleGen(), marks=[incompat])]
 
 _sortmerge_join_conf = {'spark.sql.autoBroadcastJoinThreshold': '-1',
                         'spark.sql.join.preferSortMergeJoin': 'True',
                         'spark.sql.shuffle.partitions': '2',
-                        'spark.sql.legacy.allowNegativeScaleOfDecimal': 'true'
+                        'spark.sql.legacy.allowNegativeScaleOfDecimal': 'true',
+                        'spark.rapids.sql.stableSort.enabled': 'true'
```

Comment:
Why do we need to turn this on?

Reply:
We turn on this because `GpuSortMeta` requires stable sort if there exists `StructType` in sortOrder. [link](https://github.com/NVIDIA/spark-rapids/blob/HEAD/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala#L2882)

Reply:
Then please add a comment here with a link to the issue that will fix it, so we can turn it off later. I just want to be sure that it is clear why a config is set, and that the out of core sort was not causing issues.

Reply:
I have filed an issue for that: https://github.com/NVIDIA/spark-rapids/issues/2257

Reply:
We got rid of this configuration after #2092 merged.